获取群精华消息报错

### 问题描述

1、获取群精华消息报错
在我的一个群中，其中有精华消息
```text
https://xxxx.xxxx.me
```
在调用`group.essences.getPage(0, 20)`出现异常，异常信息为
```java
java.lang.IllegalStateException: parse digest data error, status: 0 - success
	at net.mamoe.mirai.internal.contact.essence.GroupDigestProtocolKt.loadData(GroupDigestProtocol.kt:86) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.contact.essence.GroupDigestProtocolKt.getDigestList(GroupDigestProtocol.kt:107) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.contact.essence.GroupDigestProtocolKt$getDigestList$1.invokeSuspend(GroupDigestProtocol.kt) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) ~[kotlin-stdlib-1.8.20.jar:1.8.20-release-327(1.8.20)]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:138) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:112) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.access$loop(SuspendFunctionGun.kt:14) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:62) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46) ~[kotlin-stdlib-1.8.20.jar:1.8.20-release-327(1.8.20)]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:138) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:112) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.access$loop(SuspendFunctionGun.kt:14) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:62) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46) ~[kotlin-stdlib-1.8.20.jar:1.8.20-release-327(1.8.20)]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:138) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:112) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.access$loop(SuspendFunctionGun.kt:14) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:62) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46) ~[kotlin-stdlib-1.8.20.jar:1.8.20-release-327(1.8.20)]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:138) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:112) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.access$loop(SuspendFunctionGun.kt:14) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:62) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46) ~[kotlin-stdlib-1.8.20.jar:1.8.20-release-327(1.8.20)]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:138) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:112) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun.access$loop(SuspendFunctionGun.kt:14) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.deps.io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:62) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46) ~[kotlin-stdlib-1.8.20.jar:1.8.20-release-327(1.8.20)]
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.4.jar:na]
	at me.kuku.mirai.config.ScheduledSuspendMethodRunnable.run(ScheduledSuspendConfig.kt:52) ~[main/:na]
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) ~[spring-context-6.0.7.jar:6.0.7]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset$$$capture(FutureTask.java:305) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
Caused by: kotlinx.serialization.json.internal.JsonDecodingException: Failed to parse 'int'
JSON input: .....ontent":[{"msg_type":1,"text":"https://xxxxx.xxxxx.me"}]}
	at kotlinx.serialization.json.internal.JsonExceptionsKt.JsonDecodingException(JsonExceptions.kt:24) ~[kotlinx-serialization-json-jvm-1.5.0.jar:na]
	at kotlinx.serialization.json.internal.JsonExceptionsKt.JsonDecodingException(JsonExceptions.kt:32) ~[kotlinx-serialization-json-jvm-1.5.0.jar:na]
	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.unparsedPrimitive(TreeJsonDecoder.kt:146) ~[kotlinx-serialization-json-jvm-1.5.0.jar:na]
	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.decodeTaggedInt(TreeJsonDecoder.kt:380) ~[kotlinx-serialization-json-jvm-1.5.0.jar:na]
	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.decodeTaggedInt(TreeJsonDecoder.kt:36) ~[kotlinx-serialization-json-jvm-1.5.0.jar:na]
	at kotlinx.serialization.internal.TaggedDecoder.decodeIntElement(Tagged.kt:252) ~[kotlinx-serialization-core-jvm-1.5.0.jar:1.5.0]
	at net.mamoe.mirai.internal.contact.essence.DigestMessage$$serializer.deserialize(GroupDigestProtocol.kt:50) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.contact.essence.DigestMessage$$serializer.deserialize(GroupDigestProtocol.kt:50) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at kotlinx.serialization.json.internal.PolymorphicKt.decodeSerializableValuePolymorphic(Polymorphic.kt:61) ~[kotlinx-serialization-json-jvm-1.5.0.jar:na]
	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.decodeSerializableValue(TreeJsonDecoder.kt:52) ~[kotlinx-serialization-json-jvm-1.5.0.jar:na]
	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableValue(Tagged.kt:207) ~[kotlinx-serialization-core-jvm-1.5.0.jar:1.5.0]
	at kotlinx.serialization.internal.TaggedDecoder$decodeSerializableElement$1.invoke(Tagged.kt:280) ~[kotlinx-serialization-core-jvm-1.5.0.jar:1.5.0]
	at kotlinx.serialization.internal.TaggedDecoder.tagBlock(Tagged.kt:297) ~[kotlinx-serialization-core-jvm-1.5.0.jar:1.5.0]
	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableElement(Tagged.kt:280) ~[kotlinx-serialization-core-jvm-1.5.0.jar:1.5.0]
	at kotlinx.serialization.encoding.CompositeDecoder$DefaultImpls.decodeSerializableElement$default(Decoding.kt:533) ~[kotlinx-serialization-core-jvm-1.5.0.jar:1.5.0]
	at kotlinx.serialization.internal.CollectionLikeSerializer.readElement(CollectionSerializers.kt:80) ~[kotlinx-serialization-core-jvm-1.5.0.jar:1.5.0]
	at kotlinx.serialization.internal.AbstractCollectionSerializer.readElement$default(CollectionSerializers.kt:51) ~[kotlinx-serialization-core-jvm-1.5.0.jar:1.5.0]
	at kotlinx.serialization.internal.AbstractCollectionSerializer.merge(CollectionSerializers.kt:36) ~[kotlinx-serialization-core-jvm-1.5.0.jar:1.5.0]
	at kotlinx.serialization.internal.AbstractCollectionSerializer.deserialize(CollectionSerializers.kt:43) ~[kotlinx-serialization-core-jvm-1.5.0.jar:1.5.0]
	at kotlinx.serialization.json.internal.PolymorphicKt.decodeSerializableValuePolymorphic(Polymorphic.kt:61) ~[kotlinx-serialization-json-jvm-1.5.0.jar:na]
	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.decodeSerializableValue(TreeJsonDecoder.kt:52) ~[kotlinx-serialization-json-jvm-1.5.0.jar:na]
	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableValue(Tagged.kt:207) ~[kotlinx-serialization-core-jvm-1.5.0.jar:1.5.0]
	at kotlinx.serialization.internal.TaggedDecoder$decodeSerializableElement$1.invoke(Tagged.kt:280) ~[kotlinx-serialization-core-jvm-1.5.0.jar:1.5.0]
	at kotlinx.serialization.internal.TaggedDecoder.tagBlock(Tagged.kt:297) ~[kotlinx-serialization-core-jvm-1.5.0.jar:1.5.0]
	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableElement(Tagged.kt:280) ~[kotlinx-serialization-core-jvm-1.5.0.jar:1.5.0]
	at net.mamoe.mirai.internal.contact.essence.DigestList$$serializer.deserialize(GroupDigestProtocol.kt:38) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at net.mamoe.mirai.internal.contact.essence.DigestList$$serializer.deserialize(GroupDigestProtocol.kt:38) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	at kotlinx.serialization.json.internal.PolymorphicKt.decodeSerializableValuePolymorphic(Polymorphic.kt:61) ~[kotlinx-serialization-json-jvm-1.5.0.jar:na]
	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.decodeSerializableValue(TreeJsonDecoder.kt:52) ~[kotlinx-serialization-json-jvm-1.5.0.jar:na]
	at kotlinx.serialization.json.internal.TreeJsonDecoderKt.readJson(TreeJsonDecoder.kt:25) ~[kotlinx-serialization-json-jvm-1.5.0.jar:na]
	at kotlinx.serialization.json.Json.decodeFromJsonElement(Json.kt:117) ~[kotlinx-serialization-json-jvm-1.5.0.jar:na]
	at net.mamoe.mirai.internal.contact.essence.GroupDigestProtocolKt.loadData(GroupDigestProtocol.kt:84) ~[mirai-core-jvm-2.15.0-dev-90.jar:2.15.0-dev-90]
	... 44 common frames omitted
```
2、群精华消息事件是哪个？
在 #2293 中已经提及了群精华事件，该issues已经被关闭，但是我无法找到群精华消息事件。

### 复现

复现如问题描述

### mirai-core 版本

2.15.0-dev-90

### bot-protocol

ANDROID_PAD

### 其他组件版本

_No response_

### 系统日志

_No response_

### 网络日志

_No response_

### 补充信息

_No response_

--- a/mirai-core/src/commonMain/kotlin/contact/essence/EssencesImpl.kt
+++ b/mirai-core/src/commonMain/kotlin/contact/essence/EssencesImpl.kt
@@ -79,8 +79,8 @@ internal class EssencesImpl(
 
     private suspend fun source(digests: DigestMessage, parse: Boolean): MessageSource {
         return group.bot.buildMessageSource(MessageSourceKind.GROUP) {
-            ids = intArrayOf(digests.msgSeq)
-            internalIds = intArrayOf(digests.msgRandom)
+            ids = intArrayOf(digests.msgSeq.toInt())
+            internalIds = intArrayOf(digests.msgRandom.toInt())
             time = digests.senderTime
 
             fromId = digests.senderUin
@@ -122,8 +122,8 @@ internal class EssencesImpl(
     override suspend fun share(source: MessageSource): String {
         val share = group.bot.shareDigest(
             groupCode = group.id,
-            msgSeq = source.ids.first(),
-            msgRandom = source.internalIds.first(),
+            msgSeq = source.ids.first().toLong().and(0xFFFF_FFFF),
+            msgRandom = source.internalIds.first().toLong().and(0xFFFF_FFFF),
             targetGroupCode = 0
         )
         return "https://qun.qq.com/essence/share?_wv=3&_wwv=128&_wvx=2&sharekey=${share.shareKey}"
@@ -143,8 +143,8 @@ internal class EssencesImpl(
             try {
                 group.bot.cancelDigest(
                     groupCode = group.id,
-                    msgSeq = source.ids.first(),
-                    msgRandom = source.internalIds.first()
+                    msgSeq = source.ids.first().toLong().and(0xFFFF_FFFF),
+                    msgRandom = source.internalIds.first().toLong().and(0xFFFF_FFFF)
                 )
             } catch (cause: IllegalStateException) {
                 cause.addSuppressed(IllegalStateException(result.msg))

--- a/mirai-core/src/commonMain/kotlin/contact/essence/GroupDigestProtocol.kt
+++ b/mirai-core/src/commonMain/kotlin/contact/essence/GroupDigestProtocol.kt
@@ -60,9 +60,9 @@ internal data class DigestMessage(
     @SerialName("msg_content")
     val msgContent: List<JsonObject> = emptyList(),
     @SerialName("msg_random")
-    val msgRandom: Int = 0,
+    val msgRandom: Long = 0,
     @SerialName("msg_seq")
-    val msgSeq: Int = 0,
+    val msgSeq: Long = 0,
     @SerialName("sender_nick")
     val senderNick: String = "",
     @SerialName("sender_time")
@@ -108,7 +108,7 @@ internal suspend fun QQAndroidBot.getDigestList(
 }
 
 internal suspend fun QQAndroidBot.cancelDigest(
-    groupCode: Long, msgSeq: Int, msgRandom: Int
+    groupCode: Long, msgSeq: Long, msgRandom: Long
 ) {
     val data = components[HttpClientProvider].getHttpClient().get {
         url("https://qun.qq.com/cgi-bin/group_digest/cancel_digest")
@@ -133,7 +133,7 @@ internal suspend fun QQAndroidBot.cancelDigest(
 }
 
 internal suspend fun QQAndroidBot.shareDigest(
-    groupCode: Long, msgSeq: Int, msgRandom: Int, targetGroupCode: Long
+    groupCode: Long, msgSeq: Long, msgRandom: Long, targetGroupCode: Long
 ): DigestShare {
     return components[HttpClientProvider].getHttpClient().get {
         url("https://qun.qq.com/cgi-bin/group_digest/share_digest")

